Создайте метод doSomething(), который бросает RuntimeException. Обработайте исключение сразу же после броска. Вызовите этот метод из main и обработайте в нем исключение, которое вызвал метод doSomething().

Создайте класс Счетчик, у которого есть метод add(), увеличивающий значение внутренней int переменной на 1. Сделайте так, чтобы с объектом такого типа можно было работать в блоке try-with-resources и после выхода из блока, он не давал больше возможности выполнять метод add(), а при попытке это сделать бросал бы исключение. Подумайте над типом исключения.

Создайте класс исключения, который будет выбрасываться при делении на 0. Исключение должно отображать понятное для пользователя сообщение об ошибке. Создайте класс исключений, которое будет возникать при обращении к пустому элементу в массиве ссылочного типа. Исключение должно отображать понятное для пользователя сообщение об ошибке Создайте класс исключения, которое будет возникать при попытке открыть несуществующий файл. Исключение должно отображать понятное для пользователя сообщение об ошибке. 

Напишите метод, на вход которого подаётся двумерный строковый массив размером 4х4. При подаче массива другого размера необходимо бросить исключение MyArraySizeException. Далее метод должен пройтись по всем элементам массива, преобразовать в int и просуммировать. Если в каком-то элементе массива преобразование не удалось (например, в ячейке лежит символ или текст вместо числа), должно быть брошено исключение MyArrayDataException с детализацией, в какой именно ячейке лежат неверные данные. В методе main() вызвать полученный метод, обработать возможные исключения MyArraySizeException и MyArrayDataException и вывести результат расчета (сумму элементов, при условии что подали на вход корректный массив). 
